src/mesh/forest/forest.hpp

```diff
@@ -33,6 +35,46 @@
 namespace parthenon {
 namespace forest {
 
+template <class ELEMENT>
+struct ForestBC {
+  ELEMENT element;
+  BoundaryFlag bflag;
+  ELEMENT periodicElement;
+};
+
+class Forest;
+class ForestDefinition {
```

Comment:
Yeah, this is intended to be how a user would specify a forest to build a mesh from. It is obviously pretty bare bones right now, but I imagine that it will get extended in the future. For instance, we could add constructors that make standard topologies as you mention above.